// react-image-grid-gallery/dist/cjs/imageGalleryStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.imageGalleryStyles = imageGalleryStyles;
function imageGalleryStyles(columnCount, columnWidth, gapSize) {
    var galleryContainerStyle = {
        columnCount: columnCount,
        columnWidth: "".concat(columnWidth, "px"),
        columnGap: "".concat(gapSize, "px"),
    };
    var imageBtnStyle = {
        border: "none",
        background: "none",
        margin: 0,
        padding: 0,
    };
    var imageContainerStyle = {
        margin: "0 0 ".concat(gapSize, "px"),
        position: "relative",
    };
    var imageStyle = {
        display: "block",
        width: "100%",
        cursor: "pointer",
    };
    var imageCaptionStyle = {
        opacity: 0,
        transition: "opacity 1s ease-in-out",
        position: "absolute",
        bottom: 0,
        zIndex: "1000",
        width: "100%",
        backgroundColor: "rgba(0, 0, 0, 0.5)",
        background: "linear-gradient(180deg, rgba(25, 27, 38, 0), rgba(25, 27, 38, 1))",
        padding: "16px",
        fontSize: "0.85rem",
        textAlign: "center",
        color: "#fff",
    };
    var modalContainerStyle = {
        outline: "none",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        backgroundColor: "rgba(0, 0, 0, 0.73)",
        WebkitBackdropFilter: "blur(5px)",
        backdropFilter: "blur(5px)",
    };
    var modalSlideNumberStyle = {
        zIndex: 1,
        position: "absolute",
        left: 0,
        padding: "13px",
        backgroundColor: "rgba(35, 35, 35, 0.73)",
        color: "#fff",
        fontSize: "0.93rem",
        transition: "opacity 1s ease-in-out",
        userSelect: "none",
        WebkitUserSelect: "none",
    };
    var modalToolbarStyle = {
        zIndex: 1,
        display: "flex",
        position: "absolute",
        right: 0,
        alignItems: "center",
        backgroundColor: "rgba(35, 35, 35, 0.73)",
        transition: "opacity 1s ease-in-out",
        cursor: "pointer",
    };
    var modalToolbarBtnStyle = {
        margin: 0,
        border: "none",
        background: "none",
        padding: "13px",
        color: "#fff",
        cursor: "pointer",
    };
    var modalSlideShowSectionStyle = {
        position: "relative",
        display: "flex",
        alignItems: "center",
        width: "inherit",
        height: "inherit",
    };
    var modalImageStyle = {
        margin: "auto",
        maxWidth: "100vw",
        maxHeight: "100vh",
    };
    var modalSlideBtnStyle = {
        position: "absolute",
        border: "none",
        marginInline: "7px",
        padding: "10px 15px",
        backgroundColor: "rgba(35, 35, 35, 0.73)",
        color: "#fff",
        transition: "opacity 1s ease-in-out",
        cursor: "pointer",
        userSelect: "none",
        WebkitUserSelect: "none",
    };
    return {
        galleryContainerStyle: galleryContainerStyle,
        imageBtnStyle: imageBtnStyle,
        imageContainerStyle: imageContainerStyle,
        imageStyle: imageStyle,
        imageCaptionStyle: imageCaptionStyle,
        modalContainerStyle: modalContainerStyle,
        modalSlideNumberStyle: modalSlideNumberStyle,
        modalToolbarStyle: modalToolbarStyle,
        modalToolbarBtnStyle: modalToolbarBtnStyle,
        modalSlideShowSectionStyle: modalSlideShowSectionStyle,
        modalImageStyle: modalImageStyle,
        modalSlideBtnStyle: modalSlideBtnStyle,
    };
}
